### PR TITLE
Upgrade Storybook, convert stories to CSF format, implement Chromatic

### PR DESCRIPTION
**Pull request by jaronheard**
This PR upgrades Storybook 5 -> 6, and goes through the process of converting the ui-core package to use the new [Component Story Format](https://storybook.js.org/docs/react/api/csf). Besides having a great acronym, the Component Story Format enables really nice automatically generated docs, can link up with more automated testing and tools, and is much cleaner than the previous `storiesOf` notation.

It also sets up a configuration for [Chromatic](https://www.chromatic.com/), which is an advanced deploy preview and visual testing tool for Storybook. My plan would be to set it up on CI so that we get a preview on each PR.

# Feedback, please!
I'd like feedback in a few areas. 
- General comments on the new Storybook setup and component documentation
- Specific comments to improve the patterns used for stories

## ✨Check it out!✨
- [Updated Storybook with ui-core components only](https://5f55eec3d7d83100229d47fe-hilcsxsrst.chromatic.com)
- [Chromatic changes review example](https://www.chromatic.com/build?appId=5f55eec3d7d83100229d47fe&number=2)

### Notes
- The new Storybook docs & controls makes it feel better to have a large number of stories for each component. Since there's a documentation page for the components and controls are automatically generated, stories are now more like examples than the set of interactive filters that they were with knobs. I'm thinking that leaning towards a larger number of stories than our current guidelines suggest would be good in this new world.
- [react-docgen](https://github.com/reactjs/react-docgen), which Storybook uses automatically generate prop tables for Storybook Docs, does NOT like default exports of components. I had to convert components to use the default export so that the prop tables would populate. [Some](https://gitlab.com/gitlab-org/frontend/rfcs/-/issues/20) [people](https://humanwhocodes.com/blog/2019/01/stop-using-default-exports-javascript-module/) prefer this pattern but right now it breaks a eslint rule (`prefer-default-export`).
- Looks like the names of some components are getting mangled in the code samples in the built version of Storybook only. (Update: I resolved the component display names by (shocker!) adding a `displayname` to components that were missing it.)
- Speaking of code, I should include an import in the code samples, since we have components spread across modules now. (Update: This looks like it'll be [added at the storybook level soon](https://github.com/storybookjs/storybook/issues/7772))
- I used this [Storybook Tutorial](https://www.learnstorybook.com/intro-to-storybook/react/en/get-started/) for some of the best practices for writing CSF Stories. I thought it was a very helpful guide!

# Screenshots

## Docs
image
image

## The new controls view
Knobs have been replaced with controls. The type of control is inferred from the PropTypes, and the description is pulled from JSDoc comments on the PropTypes. It's pretty slick.
image

### CODE REVIEW

**Review by Dianna — APPROVED** [review]
Really great PR right up @jaronheard! This is super helpful for returning devs to understand what changed and why.

I quite like the new storybook interface and agree that both its UI and the new code component story format encourages writing more stories — which is great because these stories can both act as tests for the components and serve as documentation that's useful for designers & engineers.

I also like the new CSF. It removes a lot of nesting and makes it easy to see what code is causing what.

**Comment by jaronheard** [comment]
I've added stories for the `ui-docs` and `ui-charts` packages. One really exciting opportunity with the new Storybook MDX format is being able to mix stories and documentation. I translated the data visualization guidelines that the Data Visualization platform team from the 2019 Hack Oregon season put together into [mdx format with embedded interactive stories](https://5f55eec3d7d83100229d47fe-emmcqfjvkn.chromatic.com/iframe.html?id=design-ux-style-guide-data-visualization--page&viewMode=story). I can also imagine a tool that allows users to view their data as many different types of charts -- it would actually be very straightforward to put together!

**Comment by jaronheard** [comment]
Phew! I think that this PR is ready for final review.
- [x] ui-* Packages converted to use CSF
- [x] Component generator updated
- [x] Documentation updated
- [x] Default Storybook updated

The one exception is that `ui-maps` hasn't been converted -- I'd like to make that a follow-up issue. It also disables automatic publishing of the old Storybook. Once `ui-maps` is converted, we can update all of the links to hackoregon.github.io to the new Storybook.

I'm interested in some fresh eyes, and comparing the usability of this [updated approach](https://5f55eec3d7d83100229d47fe-ksghnpgzhf.chromatic.com/) to our [old approach](hackoregon.github.io/civic)

File: packages/component-library/package.json [hunk 1]
```diff
@@ -47,15 +47,15 @@
     "@babel/plugin-syntax-dynamic-import": "^7.0.0",
     "@babel/plugin-syntax-import-meta": "^7.0.0",
     "@babel/register": "^7.0.0",
-    "@storybook/addon-a11y": "^5.0.6",
-    "@storybook/addon-actions": "^5.0.6",
+    "@storybook/addon-a11y": "^6.0.21",
```
[the remark this answers is not in the document]
Comment (jaronheard) [reply packages/component-library/package.json:50]: Didn't revert on further consideration

**Pull request merged**